src/test/java/org/apache/commons/lang3/builder/ToStringBuilderTest.java

```diff
@@ -316,7 +316,7 @@ public void testReflectionHierarchyArrayList() {
         // representation different for IBM JDK 1.6.0, LANG-727
         assumeFalse("IBM Corporation".equals(SystemUtils.JAVA_VENDOR) && "1.6".equals(SystemUtils.JAVA_SPECIFICATION_VERSION));
         assumeFalse("Oracle Corporation".equals(SystemUtils.JAVA_VENDOR) && "1.6".compareTo(SystemUtils.JAVA_SPECIFICATION_VERSION) < 0);
-        final List<Object> list = new ArrayList<>();
+        final List<Object> list = new ArrayList<>(10);
```

Comment:
>If the test fails when the initial size arg is omitted, does that not also affect the behaviour of the method being tested?

Not really. The test simply checks the string built for an arraylist through reflection. The issue was caused for believing that the lazy initialization (as @andyklimczak) would work in the same independent of the JVM.

What the test is verifying is correct, the current approach could be improved to make the test less flaky.